### ISSUE TITLE
[terra-tabs]Accessibility Issues on terra-tabs

### ISSUE THREAD

**Issue by rohit-v** [cerner/terra-framework#1074]
# Bug Report

In the latest version of terra-tabs, a few accessibility issues were noticed while using it.

- [Chrome + Jaws], the tab  names  aren't announced.
-  When using a keyboard with more button being displayed, user must tab  to the more button before getting into the  content of the tab which isn't intuitive to a visually impaired user. (Chrome)
- Content of the tab isn't read by [JAWS in chrome], but  works in IE11.

## Steps to Reproduce
- Use the component on the page.
- Add as many tabs as possible till the button shows up.
- Try accessing it through a screen reader and through the  keyboard.

**Comment by lokesh-0813** [comment]
After investigation I found all `ARIA` attributes placed in right positions.

So I looked into the [example](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-2/tabs.html) given in [JAWS](https://freedomscientific.github.io/VFO-standards-support/aria.html) aria support. I checked the example against JAWS and it works.

My suspicion is the way we have handled "focus" in `terra-tabs` and JAWS is  not able to pick it up.

I explicitly added focus for appropriate indexes [here](https://github.com/cerner/terra-framework/blob/b6deee0d00b5da99d46f1ca8536eb6a3037b0142/packages/terra-tabs/src/_CollapsibleTabs.jsx#L218) and [here](https://github.com/cerner/terra-framework/blob/b6deee0d00b5da99d46f1ca8536eb6a3037b0142/packages/terra-tabs/src/_CollapsibleTabs.jsx#L236). And also added the pseudo "focus" classes for TabPane [here](https://github.com/cerner/terra-framework/blob/b6deee0d00b5da99d46f1ca8536eb6a3037b0142/packages/terra-tabs/src/TabPane.jsx#L65) and introduced "tabIndex" for TabPane. 

None of the changes above made JAWS say a word. I am not sure if I am looking the issue in the right direction. I could be doing something silly here. I was wondering either @dkasper-was-taken  or @tbiethman could add in some thoughts here.

**Comment by benbcai** [comment]
> - [Chrome + Jaws], the tab names aren't announced.

@rohit-v  Is this when you click on the tab or when you use the `tab` key on the keyboard to put focus on the tab? This is what I see using JAWS on Windows

- JAWS on IE11
  - Clicking on a tab reads the tab label + "tab selected". For example "Tab 5 tab selected"
  - Tabbing to put focus on the tab reads the tab label on _all_ the tabs

- JAWS on Chrome
  - Clicking on a tab does not read the tab label.
  - Tabbing to put focus on the tab reads the tab label on _all_ the tabs

![Screen Shot 2020-04-02 at 4 56 52 PM](https://user-images.githubusercontent.com/22618655/78303822-12dc4a80-7503-11ea-82d5-45b90722ac7b.png)


> - When using a keyboard with more button being displayed, user must tab to the more button before getting into the content of the tab which isn't intuitive to a visually impaired user. (Chrome)

Can you elaborate how you expect the More dropdown to work? Are you saying that once you put focus on the `More` dropdown that the labels on all the collapsed tab (the ones in the dropdown) should be announced? The collapsed tabs are announced for me only when I click or hit `Enter` to open the drop down.


> - Content of the tab isn't read by [JAWS in chrome], but works in IE11.

Is the content announced when you click on the tab in IE11? I can't get the content to read in both IE11 or Chrome when I click on the tab. I need to highlight the content text in order for it to announce.

**Comment by rohit-v** [comment]
1) The issue was with tabbing through using a keyboard.
2)  While tabbing, the expectation is  that tabbing on  the  active tab  would take you to the content of the tab. The  focus instead switches on  the more button. An additional click is needed to get to  the content.

**Comment by lokesh-0813** [comment]
@rohit-v 

> 2\. While tabbing, the expectation is  that tabbing on  the  active tab  would take you to the content of the tab. The  focus instead switches on  the more button. An additional click is needed to get to  the content.

I am not sure if tabbing from active tab should focus to tab content in the presence of `more` button. I personally wouldn't expect that to be the flow.

It makes sense to get a clear picture from the UX/ @neilpfeiffer .

**Comment by jeremyfuksa** [comment]
From @scottwilmarth:
> I quickly reviewed the tabs and there are a lot of issues that I found which are not consistent with the GitHub issue. That was only with Chrome and voiceover... not including JAWS. That said, I think it would be best to set up a meeting with Terra and Emily and me to review and discuss.

Looking for meeting time now.

**Comment by muruvetri** [comment]
@scottwilmarth  do we  have any update on this?

**Comment by jeremyfuksa** [comment]
The consensus of the meeting was that this component needs a pretty substantial overhaul from an accessibility standpoint.

**Comment by jmahurin** [comment]
Using JAWS and Chrome, I find that the right and left arrow keys intermittently change tab focus. When changing tab focus with the L/R arrow keys is locked on to one tab, using them causes JAWS to narrate each letter when you use them, continuing to do so will cause JAWS to narrate all of the letters in all of the tabs, but the focus never changes. If you use the down arrow keys, JAWS will announce each tab, but the focus does not change. Sometimes keyboard navigation works and at other times it does not. Without JAWS, the left and right arrow keys allows the user to change the tab focus.
